### PR TITLE
feat(rds): Add dbname parameter to RDS.DatabaseSecret construct

### LINKED ISSUE #24728: DatabaseSecret construct in @aws-cdk/rds should support a custom database name
### Describe the feature

RDS.DatabaseSecret construct should support setting a custom `dbname` in the secret that gets created in Secrets Manager.

### Use Case

When pulling the database-related secret from Secrets Manager, we use all of its values (username, password, database name, etc.) when creating a connection to an RDS database. However, the database name is whatever the default DB was at the time of the RDS cluster's creation, so in many cases, this means that the code pulling the secret can end up connecting to the wrong database.

### Proposed Solution

Add a variable to RDS.DatabaseSecret - PR coming soon.

### Other Information

_No response_

### Acknowledgements

- [x] I may be able to implement this feature request
- [ ] This feature might incur a breaking change

### CDK version used

2.67

### Environment details (OS name and version, etc.)

MacOS

### PR DESCRIPTION
**Pull request by hsohail94**
As explained under https://github.com/aws/aws-cdk/issues/24728, this PR adds the ability to define a custom database name in the Secrets Manager resource created by the DatabaseSecret construct under @aws-cdk/rds, so that when this secret is pulled by client code and used to set database connection parameters, it can connect to a database of its choosing, rather than using the default database created by the RDS cluster that the secret gets attached to.

Closes #24728.

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk/aws-rds/test/integ.serverless-cluster-secret-rotation-custom-names.ts.snapshot/aws-cdk-rds-integ-secret-rotation.assets.json [hunk 1]
```diff
@@ -0,0 +1,19 @@
+{
```
Comment (TheRealAmazonKendra) [inline packages/@aws-cdk/aws-rds/test/integ.serverless-cluster-secret-rotation-custom-names.ts.snapshot/aws-cdk-rds-integ-secret-rotation.assets.json:1]: I have the same question for everything here and below: was this folder manually created and/or renamed? The snapshot should have only created a `js` folder, not a `ts` one. This PR has both.

**Review by TheRealAmazonKendra — DISMISSED** [review]
Thank you for your contribution. Please see my question inline.

File: packages/@aws-cdk/aws-rds/test/integ.serverless-cluster-secret-rotation-custom-names.ts.snapshot/aws-cdk-rds-integ-secret-rotation.assets.json (hunk 1 above)
Comment (hsohail94) [reply packages/@aws-cdk/aws-rds/test/integ.serverless-cluster-secret-rotation-custom-names.ts.snapshot/aws-cdk-rds-integ-secret-rotation.assets.json:1]: Hi @TheRealAmazonKendra. This folder was generated automatically after I created my integration test, and ran it by following the integration test guidelines mentioned [here](https://github.com/aws/aws-cdk/blob/main/INTEGRATION_TESTS.md).

Comment (TheRealAmazonKendra) [reply packages/@aws-cdk/aws-rds/test/integ.serverless-cluster-secret-rotation-custom-names.ts.snapshot/aws-cdk-rds-integ-secret-rotation.assets.json:1]: Can you clarify what the exact command you ran was? It should have only created a folder with the `js` type. I also ran it locally and only that folder was created.

Comment (hsohail94) [reply packages/@aws-cdk/aws-rds/test/integ.serverless-cluster-secret-rotation-custom-names.ts.snapshot/aws-cdk-rds-integ-secret-rotation.assets.json:1]: @TheRealAmazonKendra My bash history has been cleared, but recalling from memory, I ran `yarn integ --update-on-failed` on the new test file, followed by `yarn build`.

Comment (TheRealAmazonKendra) [reply packages/@aws-cdk/aws-rds/test/integ.serverless-cluster-secret-rotation-custom-names.ts.snapshot/aws-cdk-rds-integ-secret-rotation.assets.json:1]: Did you happen to run `integ --language typescript` at some point? Either way, the `ts` folder shouldn' be merged, only the `js` one. Can you delete both snapshot folders and rerun `yarn integ --update-on-failed`?

**Review by TheRealAmazonKendra — CHANGES_REQUESTED** [review]
Putting back in changes requested to reflect status.

**Review by TheRealAmazonKendra — DISMISSED** [review]

**Comment by hsohail94** [comment]
Requested changes for integration test have been added. Please merge at your earliest convience.

**Comment by TheRealAmazonKendra** [comment]
Please don't merge from main. It stops our workflows and the merge is automatically done upon approval. This would have automatically merged when I approved it earlier today had it not been manually edited.

**Comment by TheRealAmazonKendra** [comment]
If there's something you're concerned about in the branch, you can use the `@Mergifyio update` command, but in general, you don't typically need to worry about merging from main unless you're making new changes or needing to pull in a specific change that was merged elsewhere. We've got a lot of automations that assist with stuff like this.

Also, apologies, I left the note and then forgot to re-approve.

**Review by TheRealAmazonKendra — APPROVED** [review]

**Comment by hsohail94** [comment]
> If there's something you're concerned about in the branch, you can use the `@Mergifyio update` command, but in general, you don't typically need to worry about merging from main unless you're making new changes or needing to pull in a specific change that was merged elsewhere. We've got a lot of automations that assist with stuff like this.
> 
> Also, apologies, I left the note and then forgot to re-approve.

No worries, thank you letting for me know. Based on this information, should I comment with `@Mergifyio update` if I want to keep this PR up to date with `main`? Or will your approval result in this PR being automatically kept up with `main`?

**Pull request merged**